django/contrib/admin/options.py

```diff
@@ -858,9 +858,13 @@ def action_checkbox(self, obj):
     def _get_base_actions(self):
         """Return the list of actions, prior to any request-based filtering."""
         actions = []
+        base_actions = [self.get_action(action) for action in self.actions or []]
```

Comment:
@charettes mentioned in [his review](https://github.com/django/django/pull/12591#discussion_r395680114) to avoid an unnecessary intermediary list creation. Would it be a problem?

Reply:
Actually, I prefer this implementation for readability

Reply:
We can use generator.

Reply:
Like this?

```py
base_actions = (self.get_action(action) for action in self.actions or [])
```